src/libgit2/transports/ssh.c

```diff
@@ -748,23 +750,29 @@ static int check_certificate(
 	if (check_cb != NULL) {
 		git_cert_hostkey *cert_ptr = &cert;
 		git_error_state previous_error = {0};
+		const char *host_ptr = host;
+		git_str host_and_port = GIT_STR_INIT;
+
+		if (port != SSH_DEFAULT_PORT) {
```

Comment:
_git_ssh_setup_conn sets port to -1 to indicate default but here it is compared to SSH_DEFAULT_PORT that is defined as 22.

Reply:
It does. But `git_net_url_parse` and friends should enforce that the port is a number, and set them to the default if the port number is omitted. So the `port = -1` should never be reached. As a result, I changed the `strtol` failure to an actual error. Good catch, thanks!